paramiko/hostkeys.py

```diff
@@ -117,8 +139,13 @@ def save(self, filename):
         .. versionadded:: 1.6.1
         """
         with open(filename, "w") as f:
-            for e in self._entries:
-                line = e.to_line()
+            all_entries = set(chain(*self._entries.values(), self._hashed_entries))
```

Comment:
Is there some risk of loss of information here? E.g., a collision between an otherwise identical hashed and unhashed value (this might not be possible); or, duplicate entries for a hostname (this might also not be possible)?

I *think* there's not any concern here, but my grip on the data structures is still a bit hazy so I want to raise the question.